### PR TITLE
feat: initial 0.5.0 support

### PR DESCRIPTION
**Pull request by TimoGlastra**
Needs https://github.com/Sphereon-Opensource/OID4VCI/pull/87 to properly work

Also probably want to wait for stable 0.5.0 but everything is working (with a few minor hacks)

### CODE REVIEW

**Comment by TimoGlastra** [comment]
@janrxyz I've updated everything for Credo 0.5 and also updated Expo to 50. However there seems to be something wrong with the styling now in Tamagui. And TBH I understand very little of how it all works. 

Could you maybe take a look and see why it's behaving weirdly? Last thing to do before we can release a new version

**Comment by janrxyz** [comment]
@TimoGlastra What seems to be the issue? It looks fine?

**Comment by TimoGlastra** [comment]
Did you run it? The styling is definitely off if you compare it with the version from the app store

File: packages/agent/src/display.ts [hunk 1]
```diff
@@ -194,30 +276,60 @@ export function getCredentialExchangeForDisplay(
...
+    const openId4VcMetadata = getOpenId4VcCredentialMetadata(credentialRecord)
+    const issuerDisplay = getSdJwtIssuerDisplay(openId4VcMetadata)
+    const credentialDisplay = getSdJwtCredentialDisplay(decodedPayload, openId4VcMetadata)
+
+    // TODO: should we add a way to see these claims? Currently matching what would
+    // be included in a w3c credential subject
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const { _sd_alg, _sd_hash, iss, vct, cnf, ...visibleProperties } = decodedPayload
+
+    // TODO: display somehow which fields can be selective disclosed
```
Comment (janrxyz) [inline packages/agent/src/display.ts:297]: Do we track this in code, or should we add it to a sprint board somewhere? I was testing with the wallet the other day and also saw that predicates for didcomm presentation requests are also not correct. Might be good to fix this sometime.

File: packages/agent/src/index.ts [hunk 2]
```diff
@@ -1,3 +1,13 @@
+import 'react-native-get-random-values'
+import 'fast-text-encoding'
+
+import { Buffer } from '@credo-ts/core'
+
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore
+// eslint-disable-next-line no-undef
+global.Buffer = Buffer
```
Comment (janrxyz) [inline packages/agent/src/index.ts:9]: Why is this needed?

File: packages/agent/src/parsers.ts [hunk 3]
```diff
@@ -66,71 +73,120 @@ export const receiveCredentialFromOpenId4VciOffer = async ({
...
+            }${supportedDidMethods?.join(', ') ?? 'Unknown'}`
+          )
         },
-      })
 
-      if (didResult.didState.state !== 'finished') {
-        throw new Error('DID creation failed.')
+        verifyCredentialStatus: false,
+        // FIXME: why is this removed?
+        // allowedCredentialFormats: [OpenIdCredentialFormatProfile.JwtVcJson],
```
Comment (janrxyz) [inline packages/agent/src/parsers.ts:153]: Do we know why? Can you clean this up?

**Review by janrxyz — COMMENTED** [review]
There are lots of TODOs and FIXMEs in t here, hard to judge for me which are need now or later. Maybe add clarifying comments or add them to a sprint board for later.

File: .github/workflows/continuous-integration.yaml [hunk 4]
```diff
@@ -8,11 +8,6 @@ on:
     branches:
       - main
 
-# Cancel multiple runs for the same PR or branch
```
[the remark this answers is not in the document]
Comment (TimoGlastra) [reply .github/workflows/continuous-integration.yaml:11]: No not really. It was still mentioning aries-framework so mainly a copy-pasta I think. There's nothing happening here that results in errors if multiple PRs are running at the same time

File: apps/expo/utils/walletKeyStore.ts [hunk 5]
```diff
@@ -1,21 +1,32 @@
...
   const secureStoreAvailable = await SecureStore.isAvailableAsync()
   if (!secureStoreAvailable) throw new Error('SecureStore is not available on this device.')
 
-  const walletKey = await SecureStore.getItemAsync(STORE_KEY)
-  if (walletKey) return walletKey
+  // New method: raw wallet key
+  let walletKey = await SecureStore.getItemAsync(STORE_KEY_RAW)
+  if (walletKey) return { walletKey, keyDerivation: 'raw' }
 
+  // TODO: rotate the old wallet key to a new raw key
```
[the remark this answers is not in the document]
Comment (TimoGlastra) [reply apps/expo/utils/walletKeyStore.ts:22]: This is a new todo for later. I discovered we are using uuids as the wallet key, while it's more secure and faster to generate a RAW wallet key if we're storing them in the secure enclave anyway. 

However migrating old wallets to this new method is more work, so I stopped on it as I was getting distracted with the main purpose of this PR.
Should be a ticket though

File: packages/agent/package.json [hunk 6]
```diff
@@ -4,28 +4,23 @@
...
-    "@aries-framework/anoncreds": "0.4.2",
-    "@aries-framework/anoncreds-rs": "0.4.2",
-    "@aries-framework/askar": "0.4.2",
-    "@aries-framework/cheqd": "0.4.2",
-    "@aries-framework/core": "0.4.2",
-    "@aries-framework/indy-vdr": "0.4.2",
-    "@aries-framework/react-hooks": "0.4.2",
-    "@aries-framework/react-native": "0.4.2",
-    "@internal/openid4vc-client": "*",
+    "@credo-ts/anoncreds": "0.5.1-alpha.0",
```
[the remark this answers is not in the document]
Comment (TimoGlastra) [reply packages/agent/package.json:7]: Yess bugs bugs bugs... :/

File: packages/agent/src/display.ts [hunk 7]
```diff
@@ -173,6 +215,46 @@ function getW3cCredentialDisplay(
...
+      credentialDisplay.backgroundColor = openidCredentialDisplay.background_color
+
+      if (openidCredentialDisplay.background_image) {
+        credentialDisplay.backgroundImage = {
+          url: openidCredentialDisplay.background_image.url,
+          altText: openidCredentialDisplay.background_image.alt_text,
+        }
+      }
+
+      // NOTE: logo is used in issuer display (not sure if that's right though)
```
[the remark this answers is not in the document]
Comment (TimoGlastra) [reply packages/agent/src/display.ts:240]: Both issuer and credential have a logo. We use the logo from the issuer, and the background_image from the credential.

But in that case we don't render the logo of the credential (then we could have two logos). So more a note that we don't render it at the moment

[the remark this answers is not in the document]
Comment (TimoGlastra) [reply packages/agent/src/display.ts:240]: Both issuer and credential have a logo. We use the logo from the issuer, and the background_image from the credential.

But in that case we don't render the logo of the credential (then we could have two logos). So more a note that we don't render it at the moment

File: packages/agent/src/display.ts (hunk 1 above)
Comment (TimoGlastra) [reply packages/agent/src/display.ts:297]: Ah that predicates are not correct should not happen. Can you describe what the exact issue is?

I think we should start tracking these as tickets as well yes. I'll do a pass soon to get all these tickets in

File: packages/agent/src/index.ts (hunk 2 above)
Comment (TimoGlastra) [reply packages/agent/src/index.ts:9]: We need it for cheqd support. It needs global buffer. I've added a comment

File: packages/agent/src/display.ts (hunk 1 above)
Comment (janrxyz) [reply packages/agent/src/display.ts:297]: It's this FIXME: https://github.com/animo/paradym-wallet/blob/3f7e684796a6e6a8ca033fae681f7f83fcdd55d9/packages/agent/src/hooks/useAcceptDidCommPresentation.ts#L102

File: packages/agent/src/display.ts (hunk 7 above)
[the remark this answers is not in the document]
Comment (janrxyz) [reply packages/agent/src/display.ts:240]: 👍 

Would have to think about where we place this. Is this credential logo new? Have you seen other implementations on how people render it?  It's not on our original design or sphereons (maybe they updated it).

**Review by janrxyz — APPROVED** [review]
